pilot/pkg/serviceregistry/kube/controller/ambient/statusqueue/queue.go

```diff
@@ -126,16 +146,30 @@ func Register[T StatusWriter](q *StatusQueue, name string, col krt.Collection[T]
 		patcher: func(writer StatusWriter) (kclient.Patcher, []string) {
 			return getPatcher(writer.(T))
 		},
-		start: func() {
-			col.Register(func(o krt.Event[T]) {
-				ol := o.Latest()
-				key := string(krt.GetKey(ol))
-				log.Debugf("registering key for processing: %s", key)
+		relist: func() {
+			log.Debugf("processing snapshot")
+			for _, obj := range col.List() {
```

Comment:
will this bring high qps to kube-apiserver. we have other places using such kind of queue to reconcile status like ingress and gateway

Reply:
Yeah, i mean writing status. There are other places writing status too